test/sparse/sparse.jl

```diff
@@ -1828,3 +1828,8 @@ let
     @test_throws DimensionMismatch broadcast(+, A, B, speye(N))
     @test_throws DimensionMismatch broadcast!(+, X, A, B, speye(N))
 end
+
+let A = sparse(Real[1 1])
+    @test (A + A)::SparseMatrixCSC{Int,Int} == [2 2] #19595
```

Comment:
Personally, I'd prefer the dense case to return an `Array{Int,2}`, but we'd need a revision of the behavior of arithmetic operation on dense arrays. So, checking for `eltype(A + A) === Int` here seems better to me.